### ISSUE TITLE
Using the `docker_container.image` alias results in incorrect "repo" value.

### ISSUE THREAD

**Issue by jperville** [bflad/chef-docker#507]
In version 1.0.x of this cookbook. setting the `docker_container.image` attribute would alias to the `docker_container.repo` attribute. This was useful in legacy cookbooks to keep declaring containers like this:

```ruby
docker_container 'mycontainer' do
  image '10.10.10.10/busybox'
  tag 'ubuntu-14.04'
  ...
end
```

As of version 2.0.x of this cookbook, this fails because the actual "repo" has turned internally into '10.10.10.10', resulting in incorrect (and usually non-existent) image being pulled. See end of bug description for sample stacktrace with my comment.

The difference between the 2 versions:
```sh
jp440p:chef-docker(8956m|tags/v1.0.49^0) $ fgrep -r image libraries/resource_docker_container.rb 
      alias_method :image, :repo
      alias_method :image_name, :repo
jp440p:chef-docker(8956m|tags/v1.0.49^0) $ git checkout master 
Previous HEAD position was 1877e28... updating CHANGELOG and bumping metadata
Switched to branch 'master'
Your branch is up-to-date with 'origin/master'.
jp440p:chef-docker(72m|master) $ fgrep -r image libraries/docker_container.rb
      alias_method :image_name, :image
```

As you can see, the aliases in the 2 versions above are not equivalent!

What I did was to fix my cookbook to use `docker_container.repo` instead of `docker_container.image`, but for people upgrading to version 2.0 of this cookbook this breakage still should be reported (and if possible fixed).

Sample stack trace below

```
           * docker_container[my-test-data] action create
             
             ================================================================================
             Error executing action `create` on resource 'docker_container[my-test-data]'
             ================================================================================
             
             Docker::Error::NotFoundError
             ----------------------------
       
             
             Cookbook Trace:
             ---------------
             /tmp/vagrant-cache/chef/cookbooks/docker/files/default/vendor/gems/docker-api-1.22.4/lib/docker/connection.rb:46:in `rescue in request'
             /tmp/vagrant-cache/chef/cookbooks/docker/files/default/vendor/gems/docker-api-1.22.4/lib/docker/connection.rb:38:in `request'
             /tmp/vagrant-cache/chef/cookbooks/docker/files/default/vendor/gems/docker-api-1.22.4/lib/docker/connection.rb:65:in `block (2 levels) in <class:Connection>'
             /tmp/vagrant-cache/chef/cookbooks/docker/files/default/vendor/gems/docker-api-1.22.4/lib/docker/container.rb:234:in `create'
             /tmp/vagrant-cache/chef/cookbooks/docker/libraries/docker_container.rb:175:in `block (3 levels) in <class:DockerContainer>'
       /tmp/vagrant-cache/chef/cookbooks/docker/libraries/helpers_base.rb:59:in `call'
             /tmp/vagrant-cache/chef/cookbooks/docker/libraries/helpers_base.rb:59:in `with_retries'
             /tmp/vagrant-cache/chef/cookbooks/docker/libraries/docker_container.rb:174:in `block (2 levels) in <class:DockerContainer>'
             /tmp/vagrant-cache/chef/cookbooks/compat_resource/files/lib/chef_compat/copied_from_chef/chef/provider.rb:53:in `converge_if_changed'
             /tmp/vagrant-cache/chef/cookbooks/docker/libraries/docker_container.rb:171:in `block in <class:DockerContainer>'
             
             Resource Declaration:
             ---------------------
             # In /tmp/vagrant-cache/chef/cookbooks/pm-docker/providers/data.rb
             
       51:   docker_container new_resource.name do
       52:     image helper.image_name
       53:     tag helper.image_tag
       54:     command new_resource.command
       55:     volume new_resource.volume
       
       57:   end
       58: end
             
             Compiled Resource:
             ------------------
             # Declared in /tmp/vagrant-cache/chef/cookbooks/pm-docker/providers/data.rb:51:in `deploy_container'
             
             docker_container("my-test-data") do
        action [:create]
        retries 0
        retry_delay 2
        default_guard_interpreter :default
        declared_type :docker_container
         cookbook_name "pm_docker_test"
        repo "10.0.2.2" # <--- XXXXXXXXXXXXXX broken, I initially passed 'image "10.0.2.2/busybox"
        tag "ubuntu-14.04"
        command "/bin/true"
        volumes {"/data"=>{}}
        container_name "my-test-data"
        connection #<Docker::Connection:0x00000007a15e08 @url="unix:///", @options={:socket=>"/var/run/docker.sock", "read_timeout"=>60}>
        api_version "1.20"
        network_mode "default"
        detach true
        attach_stderr true
        attach_stdout true
        log_driver "json-file"
        restart_policy "no"
        security_opts [""]
        signal "SIGKILL"
             end
```

**Comment by jperville** [comment]
Sorry for flooding you @someara with issues, but this was the last issue I found when porting my wrapper cookbook from version 1.0.49 to version 2.0.x of this excellent cookbook. I may have one more issue of container being redeployed at each chef-run that applied to version 1.0.x but I have to test first with 2.0.x to see if it still applies.

**Comment by someara** [comment]
paging @jkeiser

**Comment by chasebolt** [comment]
the image property was removed from the README back in 1.0 and added as an alias. IMO I think we should just keep it removed moving forward in 2.0 and leave it up to the user to update.

**Comment by jperville** [comment]
That's what I think also, we should just remove the alias, or raise a message inviting the user to use "repo" instead. Since version 2.0 is quite recent, users should not be too surprised.

**Comment by chasebolt** [comment]
:+1:

**Comment by jperville** [comment]
Thank you very much @chasebolt

**Comment by someara** [comment]
fix from #513 merged
